### DISCUSSION TITLE
Slides from session 4: Azure AI Search Best Practices

### DISCUSSION THREAD

**Discussion by multispark** [microsoft/AI-Chat-App-Hack#43] [Announcements]
Slides: https://speakerdeck.com/pamelafox/azure-ai-search-best-practices-for-rag-chat-apps
Video: https://www.youtube.com/watch?v=ODuDeDrs3F0

**Comment by crissins** [comment]
Thank you very much

**Comment by zzulueta** [comment]
Hello, I am trying to connect the Azure AI Search Best Practices video to your RAG Chat Application as it was unclear to me. 

Here are my questions from the video and slides.
1. The RAG Chat Repo Code uses Manual Indexing. Sri did a demo on Integrated Vectorization. Is this correct?
2. The RAG Chat Repo Code creates an Index only in Azure AI search. Sri however has an Index and a Skillset. Is this correct?
3. Is the current RAG Chat inferior because it does not have a Skillset? Or can the RAG Chat code be sufficient? 
4. In relation to 3, what would be the factors to determine if you need the skillset?
5. In relation to 3, if the RAG Chat code needs to be improved with Integrated Vectorization, how do we improve it? i.e where do we change it in the code?

**Reply by pamelafox** [reply]
1. Sri showed integrated vectorization, yes.
2. Yes, the integrated vectorization approach includes the creation of a skillset, which knows how to do splitting and vectorization.
3. The difference is that the current repo does the text splitting, and vectorization locally, whereas the skillset does it in the cloud. There may be some differences between the splitting strategies locally and in the skillset, but I'm not sure if one is clearly superior. 
4. Sri has a PR at https://github.com/Azure-Samples/azure-search-openai-demo/pull/1159 that adds the integrated vectorization setup as an option. We're still working on a couple issues with that, should be ready soon. In the meantime, the local approach should be sufficient.

**Reply by zzulueta** [reply]
Great thanks for this!
